src/shims/unix/socket.rs

```diff
@@ -76,6 +76,7 @@ impl FileDescription for SocketPair {
             // for read and write.
             epoll_ready_events.epollin = true;
             epoll_ready_events.epollout = true;
+            epoll_ready_events.epollhup = true;
```

Comment:
Please set this a few lines up, together with rdhup, to make the order of operations here make some kind of sense.